### PR TITLE
Fix c10d regression during cleanup.

### PR DESCRIPTION
**Pull request by kumpera**
This fixes a regression introduced earlier today with a change to c10d global state.

It must be cleaned up in destroy_process_group or root PG and its Store will stay alive.

Fixes regression in test_c10d_nccl.py :: RendezvousEnvTest.test_common_errors

### CODE REVIEW

**Review by H-Huang — APPROVED** [review]
LGTM

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -1143,6 +1143,8 @@ def destroy_process_group(group: Optional[ProcessGroup] = None):
         _world.pg_names.clear()
         _world.pg_group_ranks.clear()
         _world.pg_backend_config.clear()
+        _world.pg_to_tag.clear()
+        _world.tags_to_pg.clear()
 
         # when process group doesn't have an explicit name (only WORLD (default)
         # process group can have an explicit name), we use global _world.group_count
```
Comment (H-Huang) [inline torch/distributed/distributed_c10d.py:1150]: is the subgroup case also handled?

Comment (kumpera) [reply torch/distributed/distributed_c10d.py:1150]: ugg, no

**Review by wanchaol — APPROVED** [review]
Thanks for the quick fix!

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**